icicle/src/fri/fri_c_api.cpp

```diff
@@ -0,0 +1,295 @@
+  size_t commit_phase_label_len;
+
+  uint8_t* nonce_label;
+  size_t nonce_label_len;
+
+  uint8_t* public_state;
+  size_t public_state_len;
+
+  const T* seed_rng;
+};
+
+template <typename T>
+FriTranscriptConfig<T> convert_ffi_transcript_config(const FFIFriTranscriptConfig<T>* ffi_transcript_config)
+{
+  // Convert byte arrays to vectors
+  std::vector<std::byte> domain_separator_label(
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->domain_separator_label),
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->domain_separator_label) +
+      ffi_transcript_config->domain_separator_label_len);
+
+  std::vector<std::byte> round_challenge_label(
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->round_challenge_label),
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->round_challenge_label) +
+      ffi_transcript_config->round_challenge_label_len);
+
+  std::vector<std::byte> commit_phase_label(
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->commit_phase_label),
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->commit_phase_label) +
+      ffi_transcript_config->commit_phase_label_len);
+
+  std::vector<std::byte> nonce_label(
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->nonce_label),
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->nonce_label) + ffi_transcript_config->nonce_label_len);
+
+  std::vector<std::byte> public_state(
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->public_state),
+    reinterpret_cast<const std::byte*>(ffi_transcript_config->public_state) + ffi_transcript_config->public_state_len);
+
+  // Construct and return internal config
+  FriTranscriptConfig transcript_config{
+    *ffi_transcript_config->hasher_handle,
+    std::move(domain_separator_label),
+    std::move(round_challenge_label),
+    std::move(commit_phase_label),
+    std::move(nonce_label),
+    std::move(public_state),
+    *ffi_transcript_config->seed_rng};
+
+  return transcript_config;
+}
+
+extern "C" {
+typedef Hash* HasherHandle;
+
+// FriProof API
+
+typedef FriProof<scalar_t>* FriProofHandle;
+
+FriProofHandle CONCAT_EXPAND(ICICLE_FFI_PREFIX, icicle_initialize_fri_proof)() { return new FriProof<scalar_t>(); }
+
+eIcicleError CONCAT_EXPAND(ICICLE_FFI_PREFIX, icicle_delete_fri_proof)(FriProofHandle proof_ptr)
+{
+  if (!proof_ptr) return eIcicleError::INVALID_POINTER;
+  delete proof_ptr;
+  return eIcicleError::SUCCESS;
+}
+
+eIcicleError CONCAT_EXPAND(ICICLE_FFI_PREFIX, fri_proof_get_proof_sizes)(
+  FriProofHandle proof_ptr, size_t& nof_queries, size_t& nof_rounds)
+{
+  if (!proof_ptr) {
+    ICICLE_LOG_ERROR << "CANNOT USE null proof_ptr";
+    return eIcicleError::INVALID_POINTER;
+  }
+  nof_queries = proof_ptr->get_nof_fri_queries();
+  nof_rounds = proof_ptr->get_nof_fri_rounds();
+  return eIcicleError::SUCCESS;
+}
+
+eIcicleError CONCAT_EXPAND(ICICLE_FFI_PREFIX, fri_proof_get_round_proof_at)(
```

Comment:
The function name here is confusing. Sounds like a single round proof should be returned based on a round index but we are getting a pointer to all of round proofs at a specific query index
```suggestion
eIcicleError CONCAT_EXPAND(ICICLE_FFI_PREFIX, fri_proof_get_round_proofs_at)(
```

Reply:
changed to `fri_proof_get_round_proofs_for_query`